lib/ruby_smb/smb2/create_context/request.rb

```diff
@@ -0,0 +1,79 @@
+module RubySMB
+  module SMB2
+    module CreateContext
+      # [2.2.13.2.3 SMB2_CREATE_DURABLE_HANDLE_REQUEST](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/9999d870-b664-4e51-a187-1c3c16a1ae1c)
+      class CreateDurableHandleRequest < BinData::Record
+        NAME = CREATE_DURABLE_HANDLE
+
+        endian :little
+        string :reserved, label: 'Reserved', length: 16
+      end
+
+      # [2.2.13.2.11 SMB2_CREATE_DURABLE_HANDLE_REQUEST_V2](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/5e361a29-81a7-4774-861d-f290ea53a00e)
+      class CreateDurableHandleV2Request < BinData::Record
+        NAME = CREATE_DURABLE_HANDLE_V2
+
+        endian :little
+        uint32 :timeout,    label: 'Timeout'
+        struct :flags,      label: 'Flags' do
+          bit6 :reserved
+          bit1 :persistent, label: 'Persistent Handle'
+          bit1 :reserved1
+          skip length: 3
+        end
+        string :reserved,    length: 8
+        string :create_guid, label: 'Create GUID', length: 16
+      end
+
+      # [2.2.13.2.8 SMB2_CREATE_REQUEST_LEASE](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-smb2/250a5100-f8b0-4b32-a202-f592ce4c05e7)
+      class CreateLeaseRequest < BinData::Record
+        NAME = CREATE_LEASE
+
+        endian :little
+        string :lease_key,      label: 'Lease Key', length: 16
+        struct :lease_state,    label: 'Lease State' do
+          bit5 :reserved,       label: 'Reserved'
+          bit1 :read_caching,   label: 'Read Caching'
+          bit1 :handle_caching, label: 'Handle Caching'
+          bit1 :write_caching,  label: 'Write Caching'
+          skip length: 3
+        end
```

Comment:
Since there are 2 lease fields that share a tag and the type is determined by the dialect actually wiring this up is a bit more difficult. Since it's not necessary right now, I opted to remove it entirely. If leasing is ever going to be supported it'll need to be readded and that'll be a better time to ensure that it's correct.